### PR TITLE
Add DocTR to Inference

### PR DESCRIPTION
**Pull request by capjamesg**
# Description

This PR adds Grounding DINO and DocTR to Inference. This PR may be split up into multiple PRs, but is here to solicit feedback.

## Type of change

Please delete options that are not relevant.

-   [X] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

James will pair with Paul to test this PR. A limited test case of the `pip` interface is available in the root `grounding_dino.py` file.

## Any specific deployment considerations

The model weights for DocTR and Grounding DINO will need to be uploaded to Inference before use.

## Docs

Documentation is added for Grounding DINO and DocTR, although a reference needs to be added to the table of contents.

### CODE REVIEW

**Comment by capjamesg** [comment]
The DocTR model is ready for testing.

Here is an example script:

```python
from inference.models.doctr import DocTR

model = DocTR(api_key="API_KEY")

result = model.infer(
    {
        "image": {
            "type": "url",
            "value": "https://lh4.googleusercontent.com/SbSxiS2IsLLGN40SfmlXvHbkmWyrMGKODUp_YkN5Hb5NE7jT9UZtKHUkQ77BLru92UhiRmYjj1AgqRiaJ6LAIx12RRKYfsjaK79wWGQnjpntD6F3r3ApThkmu5Vy2J0CNHNqMa6YhXZE_Le_MP0gPVY",
        }
    }
)

```

File: inference/models/doctr/doctr_model.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#         "input_shape": (3, 1024, 1024),
+#         "mean": (0.798, 0.785, 0.772),
+#         "std": (0.264, 0.2749, 0.287),
+#         "url": None,
+#     },
+# }
+
+
+class DocTR(RoboflowCoreModel):
+    def __init__(self, *args, model_id: str = "doctr_rec/crnn_vgg16_bn", **kwargs):
```
Comment (paulguerrie) [inline inference/models/doctr/doctr_model.py:90]: Is model_id needed here?

**Comment by paulguerrie** [comment]
Looking good so far! Two things:

- I think we need to add a route for docTR in the http interface. 
- Also, if you plan to release DocTR on its own, can you separate it out into a separate PR (or drop the Grounding DINO stuff from this PR)?

**Comment by capjamesg** [comment]
I have removed Grounding DINO from this PR.

File: inference/models/doctr/doctr_model.py (hunk 1 above)
Comment (paulguerrie) [reply inference/models/doctr/doctr_model.py:28]: bump @capjamesg

File: inference/core/interfaces/http/http_api.py [hunk 2]
```diff
@@ -612,6 +613,44 @@ async def clip_compare(
                         trackUsage(clip_model_id, actor, n=2)
                     return response
 
+            if CORE_MODEL_DOCTR_ENABLED:
+                @app.post(
+                    "/doctr/ocr",
+                    response_model=M.DoctrOCRInferenceResponse,
```
Comment (paulguerrie) [inline inference/core/interfaces/http/http_api.py:619]: `data_models as M` is no longer imported in this file. You'll need to add imports for each request and response object individually from `inference/core/entities`

**Comment by paulguerrie** [comment]
Looks like you need to run `make style` and commit the changes.

**Comment by paulguerrie** [comment]
Last thing I just realized this needs is a test. You can follow `inference/tests/inference/integration_tests/gaze_test.py` as an example. Other than that, looking good!

**Comment by capjamesg** [comment]
I have added a test case and fixed a bug in the inference logic.

**Comment by paulguerrie** [comment]
> I have added a test case and fixed a bug in the inference logic.

I think your last commit did not include a test. Did you maybe forget to add the file before committing?

**Review by paulguerrie — APPROVED** [review]
Looks good assuming regression tests pass!

**Pull request merged**